IBAnimatable/AnimationChainable.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  Created by jason akakpo on 01/01/2017.
+//  Copyright © 2017 IBAnimatable. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+typealias AnimationTuple = (type: AnimationType, configuration: AnimationConfiguration)
+
+struct AnimationConfiguration {
+  let damping: CGFloat
+  let velocity: CGFloat
+  let duration: TimeInterval
+  let delay: TimeInterval
+  let force: CGFloat
+
+}
+
+public class AnimationPromise<T: UIView> where T: Animatable {
+
+  private var view: T
+  private var animationList = [AnimationTuple]()
+  private var delayForNextAnimation = 0.0
+  private var completion: AnimatableCompletion?
+
+  init(view: T) {
+    self.view = view
+  }
+
+   func animationCompleted() {
+      animationList.remove(at: 0)
+      if let anim = animationList.first {
+        view.doAnimation(anim.type, configuration: anim.configuration, promise: self)
+      } else {
+        completion?()
+    }
+  }
+
+  public func completion(_ completion: AnimatableCompletion?) {
+    self.completion = completion
+  }
+
+  @discardableResult
+  public func  then(_ animation: AnimationType, duration: TimeInterval? = nil, damping: CGFloat? = nil, velocity: CGFloat? = nil, force: CGFloat? = nil) -> AnimationPromise {
+    let animTuple = AnimationTuple(type: animation, configuration: AnimationConfiguration(damping: damping ?? view.damping, velocity: velocity ?? view.velocity, duration: duration ?? view.duration, delay: delayForNextAnimation, force: force ?? view.force))
+    animationList.append(animTuple)
+    if animationList.count == 1 { // If it's the only animation, launch it immediately
+      view.doAnimation(animation, configuration: animTuple.configuration, promise: self)
```

Comment:
I was thinking something like
```
view.delay(2)
.animate(.pop(repeatCount: 1))
```

Is it possible to delay the first animations?